### DISCUSSION TITLE
Paged search examples

### DISCUSSION THREAD

**Discussion by chrisrrowland** [dsbenghe/Novell.Directory.Ldap.NETStandard#188] [General]
I have tried using both SimplePagedResultsControlHandler and VirtualListViewControlHandler and my expectation was that each of  these would return list with the entirety of the search  results.  What I'm actually getting in both cases is a  list with 1500 elements, which as I understand is the max result size for the server.

```
                ////SimplePagedResultsControlHandler
                var pageControlHandler = new SimplePagedResultsControlHandler(conn);
                List<LdapEntry> rows = pageControlHandler.SearchWithSimplePaging(options, pageSize: 1000);
```

```
                //VirtualListViewControlHandler
                var vlvControlHandler = new VirtualListViewControlHandler(conn);
                var sortControl = new LdapSortControl(new LdapSortKey("cn"), true);
                List<LdapEntry> rows = vlvControlHandler.SearchUsingVlv(sortControl, options, 1000);
```

**Comment by dsbenghe** [comment]
Tests are the best example currently.
Don't know from where your 1500 is coming, but I have run it with a lot larger results.

**Comment by chrisrrowland** [comment]
It turns out my issue was that I was getting back a single _object_ but the _attribute_ list was too large.  I ended up manually paging the  attributes by tweaking the request.  I'm not sure if this is something weird in the LDAP world or if it's something this library could handle automatically and I just didn't see how.  

I have very limited understanding of ldap so it's even difficult to describe my issue :-p
